### PR TITLE
chore: Project as a Boundary dataset integration tests

### PR DESCRIPTION
**Pull request by aws-tyler**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-app/src/dataSetRoutes.ts [hunk 1]
```diff
@@ -209,7 +206,7 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
   );
 
   router.delete(
-    '/projects/:projectId/datasets/:datasetId',
+    '/projects/:projectId/datasets/:datasetId/softDelete',
```
Comment (bhudnell) [inline solutions/swb-app/src/dataSetRoutes.ts:209]: `/softDelete` is gone now right?

File: solutions/swb-reference/integration-tests/support/resources/datasets/dataset.ts [hunk 2]
```diff
@@ -18,24 +18,24 @@ export default class Dataset extends Resource {
...
-    return this._axiosInstance.put(`/projects/${projectId}/datasets/${this._id}/relationships`, {
+    return this._axiosInstance.put(`/projects/${associatedProjectId}/datasets/${this._id}/relationships`, {
       accessLevel
     });
   }
 
-  public async deleteFromProject(projectId: string): Promise<void> {
-    return this._axiosInstance.delete(`/projects/${projectId}/datasets/${this._id}`);
+  public async softDelete(): Promise<AxiosResponse> {
+    return this._axiosInstance.delete(`${this._api}/softDelete`);
```
Comment (bhudnell) [inline solutions/swb-reference/integration-tests/support/resources/datasets/dataset.ts:30]: same comment about `softDelete`

File: common/changes/@aws/workbench-core-accounts/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-accounts/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json:1]: doesnt look like accounts got touched

File: common/changes/@aws/workbench-core-datasets/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-datasets/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json:2]: doesnt look like datasets got touched

File: common/changes/@aws/workbench-core-environments/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-environments/atmikev-paab-dataset-integration-tests_2023-02-25-00-36.json:3]: doesnt look like environments got touched

**Review by bhudnell — APPROVED** [review]

**Review by kpark277 — APPROVED** [review]

**Pull request merged**